spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java

```diff
@@ -0,0 +1,360 @@
+				.buildAndExpand(uriVariables).encode().toUri();
+		return this.restTemplate.getForObject(uri, returnType);
+	}
+
+	public static List<Policy> policyMapsToPolicies(List<Map<String, Object>> policies) {
+		List<Policy> list = new ArrayList<Policy>(policies.size());
+		for (Map<String, Object> map : policies) {
+			list.add(policyMapToPolicy(map));
+		}
+		return list;
+	}
+
+	@SuppressWarnings("unchecked")
+	public static Policy policyMapToPolicy(Map<String, Object> map) {
+		return new Policy((String) map.get("vhost"), (String) map.get("name"),
+				(String) map.get("pattern"), (String) map.get("apply_to"), (Integer) map.get("priority"),
+				(Map<String, Object>) map.get("definition"));
+	}
+
+	public static List<Exchange> exchangeMapsToExchanges(List<Map<String, Object>> exchangeMaps) {
+		List<Exchange> exchanges = new ArrayList<Exchange>(exchangeMaps.size());
+		for (Map<String, Object> map : exchangeMaps) {
+			exchanges.add(exchangeMapToExchange(map));
+		}
+		return exchanges;
+	}
+
+	public static Exchange exchangeMapToExchange(Map<String, Object> map) {
+		AbstractExchange exchange = null;
+		final String type = (String) map.get("type");
+		String name = (String) map.get("name");
+		boolean durable = (Boolean) map.get("durable");
+		boolean autoDelete = (Boolean) map.get("auto_delete");
+		@SuppressWarnings("unchecked")
+		Map<String, Object> arguments = (Map<String, Object>) map.get("arguments");
+		map.remove("type");
+		map.remove("name");
+		map.remove("durable");
+		map.remove("auto_delete");
+		map.remove("arguments");
+		if (type.equals("direct")) {
+			exchange = new DirectExchange(name, durable, autoDelete, arguments, map);
+		}
+		else if (type.equals("topic")) {
+			exchange = new TopicExchange(name, durable, autoDelete, arguments, map);
+		}
+		else if (type.equals("fanout")) {
+			exchange = new FanoutExchange(name, durable, autoDelete, arguments, map);
+		}
+		else if (type.equals("headers")) {
+			exchange = new HeadersExchange(name, durable, autoDelete, arguments, map);
+		}
+		else {
+			class UnknownExchange extends AbstractExchange {
+
+				public UnknownExchange(String name, boolean durable, boolean autoDelete, Map<String, Object> arguments) {
+					super(name, durable, autoDelete, arguments);
+				}
+
+				@Override
+				public String getType() {
+					return type;
+				}
+
+			}
+			exchange = new UnknownExchange(name, durable, autoDelete, arguments);
+		}
+		return exchange;
+	}
+
+	public static List<Queue> queueMapsToQueues(List<Map<String, Object>> queueMaps) {
+		List<Queue> queues = new ArrayList<Queue>(queueMaps.size());
+		for (Map<String, Object> map : queueMaps) {
+			queues.add(queueMapToQueue(map));
+		}
+		return queues;
+	}
+
+	public static Queue queueMapToQueue(Map<String, Object> map) {
+		String name = (String) map.get("name");
+		boolean durable = (Boolean) map.get("durable");
+		boolean exclusive = false; // not returned (Boolean) map.get("exclusive");
```

Comment:
Yeah, pretty much everything reporting-ish in the API is delayed, we periodically emit stats events (and `exclusive_consumer_tag` counts as a "statistic" internally).

I'm saying that if you declare an exclusive queue (not a queue with an exclusive consumer) then it will have an `owner_pid_details` key in the queue JSON. There's no `exclusive` key.